agent/src/beerocks/slave/CMakeLists.txt

```diff
@@ -31,7 +31,7 @@ endif()
 
 add_executable(${PROJECT_NAME} ${beerocks_agent_sources})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-Wl,-z,defs")
-target_link_libraries(${PROJECT_NAME} rt dl bcl btl tlvf elpp btlvf ${LIBS})
+target_link_libraries(${PROJECT_NAME} rt dl bcl btl tlvf elpp btlvf bwl ${LIBS})
```

Comment:
I'm confused - how did it work without it? What was the -I path before this change?

Reply:
target_link_libraries automatically pulls in the interface properties (for an external dependency) resp. the public properties (for an internal dependency). Thus, the PRIVATE include_directories property of beerocks slave will be updated with the PUBLIC include_directories property of bwl after this change.

Before this change, bwl headers were installed by the common module, and this module would just find them in the install/include directory that is added explicitly in each CMakeLists.txt file.

Reply:
But what about the fact that the slave uses dlopen to load BWL? Won't this mean also that BWL will be dynamically linked and there would be no need for dlopen?

Reply:
It is true that with this change, the slave will be linked with bwl, while before bwl was only loaded with dlopen().

But I don't see the use of using dlopen() here. It's not really a plugin, because you still need to give a CMake option to say which bwl to build. And even without dlopen, you can at runtime replace the .so with a different .so that exports the same functions, so I don't see the point. dlopen() would only be useful if you could choose at runtime which bwl to use - but I don't see why that is useful.